docs/howtos/model_surgery.rst

```diff
@@ -0,0 +1,232 @@
+Model Surgery
+==============================
+
+.. testsetup::
+
+  import functools
+  import numpy as np
+  import jax
+  from jax import lax, random, numpy as jnp
+  import flax
+  from flax import optim, traverse_util
+
+  from flax import linen as nn
+  from flax.core import unfreeze, freeze
+
+We will show how to get a flat dict of all the tensors, and then go back to a 
+nested, frozen dict. We show this for both Flax modules and optimizers.
+
+Surgery with Flax Modules
+--------------------------------
+
+Let's first make a little model for our demo.
+
+.. testcode::
+
+  class CNN(nn.Module):
```

Comment:
Why this particular structure?
Did you consider making it shorter not to distract from the more important code?
(Or maybe adding another level of modules if you think that's insightful)

Reply:
It's the default CNN that's used in many of our HOWTOs. We could change it, but I actually think it is quite nice since it simply yet realistic.